### ISSUE TITLE
Apps: answers.yaml not in sync with "Edit as YAML" yaml

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#20087]
Version: Master (v2.3) (5/8/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create an app, Edit as YAML, plug in some YAML to supply answers
- View answers.yaml

**Result:**
answers.yaml shows my answers as null but this is not true. My YAML provided the answers values.
Note this doesn't break anything it's just a rendering issue in the UI. The app will deploy correctly.

**Other details that may be helpful:**
In my case I tested in a specific way:

Catalog used:
URL: https://github.com/leodotcloud/rancher-charts.git
Branch: authelia
App used: Authelia

Supply this YAML when "Edit as YAML"
```
authelia:
  enableHealtcheck: false
  config:
    domain: "example.com"
    logsLevel: "debug"
    session:
      secret: "changeme"
    authentication_backend: "ldap"
    ldap:
      url: "ldap://ldap.example.com"
      user: "cn=admin,dc=example,dc=com"
      password: "admin"
      base_dn: "dc=example,dc=com"
      additional_users_dn: ""
      users_filter: "cn={0}"
      additional_groups_dn: ""
      groups_filter: "(&(member={dn})(objectclass=group))"
      group_name_attribute: "cn"
      mail_attribute: "mail"
  ingress:
    enabled: true
    hosts:
      - "login.example.com"
```
View answers.yaml and it shows entries I provided a value with as null.

This does NOT break the app - the app will deploy correctly with correct valuesYAML data. This is just a rendering issue -- the UI just isn't show the answers.yaml as we would expect but it doesn't hurt anything.

**Comment by jiaqiluo** [comment]
the bug fix is validated in Rancher:master-head 8d57a1cfb with UI index `https://releases.rancher.com/ui/latest2/index.html`

Used the same app and values from the original issue, and now the values are synced between the `edit as yaml` and `edit as form` modes when switching between them. Also, loading values from files also work and the values are sync between those two modes as well.

**Comment by deniseschannon** [comment]
@jiaqiluo This was merged into the UI before 2.3 shipped so even though it's 1 PR it's available in 2.3-head and master.

**Comment by jiaqiluo** [comment]
the bug fix is validated in Rancher:v2.3-head with UI index `https://releases.rancher.com/ui/latest-2.3/index.html`

Used the same app and values from the original issue, and now the values are synced between the `edit as yaml` and `edit as form` modes when switching between them. Also, loading values from files also work and the values are sync between those two modes as well.
